consai_robot_controller/src/field_info_parser.cpp

```diff
@@ -725,97 +725,120 @@ bool FieldInfoParser::avoid_obstacles(
 
   const double VISIBILITY_THRESHOLD = 0.01;  // 0.0 ~ 1.0
   // 自身から直進方向に何m離れたロボットを障害物と判定するか
-  const double OBSTACLE_DETECTION_X = 0.2;
+  const double OBSTACLE_DETECTION_X = 0.5;
   // 自身から直進方向に対して左右何m離れたロボットを障害物と判定するか
   const double OBSTACLE_DETECTION_Y = 0.5;
   // 相対的な回避位置
-  const double AVOIDANCE_POS_X = 0.2;
-  const double AVOIDANCE_POS_Y = 0.4;
+  double avoidance_pos_x = 0.0;
+  double avoidance_pos_y = 0.0;
+
+  // ボール回避判定フラグ
+  bool is_avoid = false;
 
   auto my_robot_pose = tools::pose_state(my_robot);
   tools::Trans trans_MtoG(my_robot_pose, tools::calc_angle(my_robot_pose, goal_pose));
+  auto my_robot_pose_MtoG = std::make_shared<State>(trans_MtoG.transform(my_robot_pose));
   auto goal_pose_MtoG = trans_MtoG.transform(goal_pose);
+  auto avoidance_pose_MtoG = goal_pose_MtoG;
 
-  double distance_to_obstacle = 10000;  // 適当な大きい数字を設定
+  double distance = 0.0; // 距離を格納する変数
+  double distance_to_obstacle = 10000; // 適当な大きい数字を設定
   std::shared_ptr<State> obstacle_pose_MtoG;
 
-  for (const auto & robot : detection_tracked_->robots) {
-    if (robot.visibility.size() == 0) {
-      continue;
-    }
-    if (robot.visibility[0] < VISIBILITY_THRESHOLD) {
-      continue;
-    }
+  int i;
+  for (i = 0; i < 6; i++) {
```

Comment:
同じ処理を繰り返し実行することがわかるように、明記したいです。

例：
```suggestion
  // 回避位置の計算を反復することで、回避位置生成後の回避位置-自己位置間の障害物も検索できる
  for (auto iter = 0; iter < NUM_ITERATIONS; iter++) {
```

Reply:
メモ：書いた